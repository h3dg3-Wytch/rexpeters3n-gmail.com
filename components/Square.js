import React, { useState } from "react";
import ReactDOM from "react-dom";
import { TouchableOpacity, Text, StyleSheet } from 'react-native';


function Square({ value, onPress }) {
  return (
    <TouchableOpacity style={styles.square} onPress={onPress}>
      <Text style={styles.squareText}>{value}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
    square: {
       backgroundColor: '#add8e6',
       padding: 10,
       width: '136px',
       height: '136px',
       margin: '4px',
       display: 'flex',
       justifyContent: 'center',
       alignItems: 'center',
    },
    squareText: {
        fontSize: '40px',
    }
});
export default Square;