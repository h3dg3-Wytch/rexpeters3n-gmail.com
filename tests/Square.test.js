import React from 'react';
import renderer from 'react-test-renderer';

import Square from '../components/Square';

import { Text } from 'react-native';

describe('<Square />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<Square/>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children.length).toBe(1);
  });

  it('can take the value X', () => {
    const testInstance = renderer.create(<Square value={'X'}/>).toJSON();
    expect(testInstance.children[0].children[0]).toEqual('X');
  });

});