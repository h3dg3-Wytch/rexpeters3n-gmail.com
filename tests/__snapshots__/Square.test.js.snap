// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> has 1 child 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#add8e6",
      "display": "flex",
      "height": 120,
      "justifyContent": "center",
      "margin": 4,
      "opacity": 1,
      "padding": 10,
      "width": 120,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 40,
      }
    }
  />
</View>
`;
